core/src/main/java/org/apache/iceberg/TableMetadataParser.java

```diff
@@ -561,4 +576,21 @@ private static List<StatisticsFile> statisticsFilesFromJson(JsonNode statisticsF
 
     return statisticsFilesBuilder.build();
   }
+
+  private static List<PartitionStatisticsFile> partitionStatisticsFilesFromJson(
+      JsonNode partitionStatisticsFilesList) {
```

Comment:
I don't mind using longer variables but I wonder whether a shorter version and staying on one line in some places would be more readable in this method.